From 98d4448ce8b2165206590ef3cc74a0ba81b806dd Mon Sep 17 00:00:00 2001
From: Akshay Gupta <Akshay.Gupta@amd.com>
Date: Thu, 3 Apr 2025 04:16:24 -0500
Subject: [PATCH 2/6] sbrmi: update label for hwmon entry to include socket
 number

The data name label under hwmon for i2c is sbrmi and for i3c is sbrmi_i3c.
This makes difficult to distinguish between socket 0 and socket 1 from
hwmon entry. New name for i2c/i3c will be sbrmi_X.Y, where X is socket
index and Y is IOD index in the socket
Signed-off-by: Akshay Gupta <Akshay.Gupta@amd.com>
Signed-off-by: Naveen Krishna Chatradhi <naveenkrishna.chatradhi@amd.com>
---
 drivers/apml/sbrmi.c | 40 +++++++++++++++++++++++++++++++++-------
 1 file changed, 33 insertions(+), 7 deletions(-)

diff --git a/drivers/apml/sbrmi.c b/drivers/apml/sbrmi.c
index 03e767969..bae96c1ee 100644
--- a/drivers/apml/sbrmi.c
+++ b/drivers/apml/sbrmi.c
@@ -402,6 +402,7 @@ static int sbrmi_i2c_probe(struct i2c_client *client)
 	struct device *dev = &client->dev;
 	struct device *hwmon_dev;
 	struct apml_sbrmi_device *rmi_dev;
+	const char *name;
 
 	rmi_dev = devm_kzalloc(dev, sizeof(struct apml_sbrmi_device), GFP_KERNEL);
 	if (!rmi_dev)
@@ -414,7 +415,21 @@ static int sbrmi_i2c_probe(struct i2c_client *client)
 
 	dev_set_drvdata(dev, (void *)rmi_dev);
 
-	hwmon_dev = devm_hwmon_device_register_with_info(dev, client->name,
+	rmi_dev->dev_static_addr = client->addr;
+
+	switch(rmi_dev->dev_static_addr) {
+	case 0x3c:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_%s", "0.0");
+		break;
+	case 0x38:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_%s", "1.0");
+		break;
+	default:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_");
+		break;
+	}
+
+	hwmon_dev = devm_hwmon_device_register_with_info(dev, name,
 							 rmi_dev,
 							 &sbrmi_chip_info,
 							 NULL);
@@ -422,8 +437,6 @@ static int sbrmi_i2c_probe(struct i2c_client *client)
 	if (!hwmon_dev)
 		return PTR_ERR_OR_ZERO(hwmon_dev);
 
-	rmi_dev->dev_static_addr = client->addr;
-
 	init_completion(&rmi_dev->misc_fops_done);
 	return create_misc_rmi_device(rmi_dev, dev);
 }
@@ -563,6 +576,7 @@ static int sbrmi_i3c_probe(struct i3c_device *i3cdev)
 	struct device *dev = &i3cdev->dev;
 	struct device *hwmon_dev;
 	struct apml_sbrmi_device *rmi_dev;
+	const char *name;
 
 	rmi_dev = devm_kzalloc(dev, sizeof(struct apml_sbrmi_device), GFP_KERNEL);
 	if (!rmi_dev)
@@ -575,15 +589,27 @@ static int sbrmi_i3c_probe(struct i3c_device *i3cdev)
 
 	dev_set_drvdata(dev, (void *)rmi_dev);
 
-	hwmon_dev = devm_hwmon_device_register_with_info(dev, "sbrmi_i3c", rmi_dev,
+	/* Need to verify for the static address for i3cdev */
+	rmi_dev->dev_static_addr = i3cdev->desc->info.static_addr;
+
+	switch(rmi_dev->dev_static_addr) {
+	case 0x3c:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_%s", "0.0");
+		break;
+	case 0x38:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_%s", "1.0");
+		break;
+	default:
+		name = devm_kasprintf(dev, GFP_KERNEL, "sbrmi_");
+		break;
+	}
+
+	hwmon_dev = devm_hwmon_device_register_with_info(dev, name, rmi_dev,
 							 &sbrmi_chip_info, NULL);
 
 	if (!hwmon_dev)
 		return PTR_ERR_OR_ZERO(hwmon_dev);
 
-	/* Need to verify for the static address for i3cdev */
-	rmi_dev->dev_static_addr = i3cdev->desc->info.static_addr;
-
 	init_completion(&rmi_dev->misc_fops_done);
 	return create_misc_rmi_device(rmi_dev, dev);
 }
-- 
2.43.0

